# Load Data

In [0]:
!wget https://www.mlgraz.at/course_files/week3/avocado.csv

--2020-04-07 18:08:13--  https://www.mlgraz.at/course_files/week3/avocado.csv
Resolving www.mlgraz.at (www.mlgraz.at)... 81.19.159.92
Connecting to www.mlgraz.at (www.mlgraz.at)|81.19.159.92|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1989197 (1.9M) [text/csv]
Saving to: ‘avocado.csv.1’

avocado.csv.1       100%[===================>]   1.90M  1.73MB/s    in 1.1s    

2020-04-07 18:08:15 (1.73 MB/s) - ‘avocado.csv.1’ saved [1989197/1989197]



In [0]:
import pandas as pd
import numpy as np
import plotly.express as px

In [0]:
df = pd.read_csv('avocado.csv')

In [0]:
df.Date = pd.to_datetime(df.Date)

In [0]:
print('Mean: ',df['4770'].mean())
print('Median: ',df['4770'].median())
print('std: ',df['4770'].std())
print('min: ',df['4770'].min())
print('max: ',df['4770'].max())

Mean:  22839.735992657315
Median:  184.99
std:  107464.06843537689
min:  0.0
max:  2546439.11


In [0]:
6.250565e+07

62505650.0

In [0]:
df['Total Volume'].median()

107376.76

# Histogram

**make a Histogram of prices for conventional avocados**

In [0]:
df_conventional = df[df.type == 'conventional']

In [0]:
fig = px.histogram(data_frame=df_conventional, x="AveragePrice", title='Distribution of prices on Conventional Avocado')
fig.show()

**make a Histogram of prices for organic avocados**
Q



In [0]:
df_organic = df[df.type == 'organic']

In [0]:
fig = px.histogram(data_frame=df_organic, x="AveragePrice", title='Distribution of prices on Organic Avocado')
fig.show()

**Stack Conventional and Organic Histogram Plots**

In [0]:
fig = px.histogram(df, x="AveragePrice", color="type")
fig.show()

## Boxplot

**Make a boxplot for conventional Avocados**

In [0]:
fig = px.box(df_conventional, y="AveragePrice")
fig.show()

**Create a boxplot for the prices of the organic avocados** Q

In [0]:
fig = px.box(df_organic, y="AveragePrice")
fig.show()

**Create a stacked boxplot for all the types of avocados**

In [0]:
fig = px.box(df, y="AveragePrice", x='type')
fig.show()

# Lineplot

**Plot line plot (Date and Price) for conventional avocados**

In [0]:
fig = px.line(df[df.type == 'conventional'], x="Date", y="AveragePrice")
fig.show()

**What is Groupby?**


In [0]:
df.groupby("type").mean()

Unnamed: 0  AveragePrice  ...  XLarge Bags         year
type                                    ...                          
conventional   24.236686      1.158040  ...  6210.635065  2016.147929
organic        24.227776      1.653999  ...     1.197164  2016.147868

[2 rows x 11 columns]

In [0]:
df.groupby("type").AveragePrice.mean()

type
conventional    1.158040
organic         1.653999
Name: AveragePrice, dtype: float64

In [0]:
df.groupby("type").AveragePrice.min()

type
conventional    0.46
organic         0.44
Name: AveragePrice, dtype: float64

In [0]:
df.groupby("type").AveragePrice.min().reset_index()

type  AveragePrice
0  conventional          0.46
1       organic          0.44

In [0]:
df.groupby(['Date', 'type']).AveragePrice.mean()

Date        type        
2015-01-04  conventional    1.009630
            organic         1.592963
2015-01-11  conventional    1.111296
            organic         1.630000
2015-01-18  conventional    1.133519
                              ...   
2018-03-11  organic         1.533519
2018-03-18  conventional    1.096296
            organic         1.531111
2018-03-25  conventional    1.147222
            organic         1.546481
Name: AveragePrice, Length: 338, dtype: float64

In [0]:
df.groupby(['Date', 'type']).AveragePrice.mean().reset_index()

Date          type  AveragePrice
0   2015-01-04  conventional      1.009630
1   2015-01-04       organic      1.592963
2   2015-01-11  conventional      1.111296
3   2015-01-11       organic      1.630000
4   2015-01-18  conventional      1.133519
..         ...           ...           ...
333 2018-03-11       organic      1.533519
334 2018-03-18  conventional      1.096296
335 2018-03-18       organic      1.531111
336 2018-03-25  conventional      1.147222
337 2018-03-25       organic      1.546481

[338 rows x 3 columns]

**Create a Line plot (Date, AveragePrice). Take the mean of entry, grouped by date and type. Plot only for conventional avocados**

In [0]:
df_mean_price = df.groupby(['Date', 'type']).AveragePrice.mean().reset_index()

In [0]:
fig = px.line(df_mean_price[df_mean_price.type == 'conventional'], x="Date", y="AveragePrice")
fig.show()

**Ok, now do the same for organic.** q

In [0]:
fig = px.line(df_mean_price[df_mean_price.type == 'organic'], x="Date", y="AveragePrice")
fig.show()

**Finally, create a stacked line plot for both types**

In [0]:
fig = px.line(df_mean_price, x="Date", y="AveragePrice",  color='type')
fig.show()

# Barplot

**Create a df "df_2017" where you take all value in the year 2017** Q

Tip: Use column name "year"

In [0]:
df_2017 = df[df.year == 2017]
df_2017.shape


(5722, 14)

In [0]:
df_2017 = df[df.Date.dt.year == 2017]
df_2017.shape

(5722, 14)

How many rows are there in df_2017?

5722

**Now Groupby "region" and "type" on df_2017. Calculate the mean of the "AveragePrice" column. Reset the index and save it in a variable called "df_2017_region_price".**

In [0]:
df_2017_region_price = df_2017.groupby(['region', 'type']).AveragePrice.mean().reset_index()
df_2017_region_price

region          type  AveragePrice
0                 Albany  conventional      1.526415
1                 Albany       organic      1.749245
2                Atlanta  conventional      1.192453
3                Atlanta       organic      1.665094
4    BaltimoreWashington  conventional      1.514717
..                   ...           ...           ...
103              TotalUS       organic      1.651509
104                 West  conventional      1.098302
105                 West       organic      1.681698
106     WestTexNewMexico  conventional      0.903585
107     WestTexNewMexico       organic      1.674314

[108 rows x 3 columns]

**How many rows and columns does "df_2017_region_price" have?**

108,3

In [0]:
df_2017_region_price.sort_values(by=['type', 'AveragePrice'],ascending=True, inplace=True)

In [0]:
city_name = ['NewYork', 'LosAngeles', 'Chicago','Houston', 'Denver', 'Philadelphia', 'Seattle', 'SanDiego', 'SanFrancisco', 'Seattle']

**Create a new df with only the cities in "city_name"**

In [0]:
df_2017_city_price = df_2017_region_price[df_2017_region_price.region.isin(city_name)]

**Plot a barchart ("region" and "AveragePrice") with the new data frame. Plot only from the conventional avocados.**

In [0]:
fig = px.bar(df_2017_city_price[df_2017_city_price.type == 'conventional'], y='region', x='AveragePrice', orientation='h')
fig.show()

In [0]:
fig = px.bar(df_2017_city_price[df_2017_city_price.type == 'conventional'], x='region', y='AveragePrice', orientation='v')
fig.show()

**Now do the same with the organic ones**

In [0]:
fig = px.bar(df_2017_city_price[df_2017_city_price.type == 'organic'], y='region', x='AveragePrice', orientation='h')
fig.show()

**Now stack the two barplots**

In [0]:
df_2017_city_price.sort_values(by=['AveragePrice'],ascending=False, inplace=True)
fig = px.bar(df_2017_city_price, x='region', y='AveragePrice', color='type',barmode='group')
fig.show()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



# Scatterplot

In [0]:
df_sf = df[(df.region == 'SanFrancisco')]

**Make a scatter plot ("AveragePrice",  "Total Volume") for the conventional avocados in San Francisco**

In [0]:
fig = px.scatter(df_sf[df_sf.type == 'conventional'], x='AveragePrice', y='Total Volume')
fig.show()

**Make a scatter plot ("AveragePrice",  "Total Volume") for the organic avocados in Houston**

In [0]:
df_h = df[(df.region == 'Houston')]

In [0]:
fig = px.scatter(df_h[df_h.type == 'organic'], x='AveragePrice', y='Total Volume')
fig.show()

**Stack the Houston and San Francisco Scatterplots (organic avocados)**

In [0]:
df_sf_h = df[(df.region.isin(['SanFrancisco', 'Houston'])) & (df.type == 'organic')]

In [0]:
fig = px.scatter(df_sf_h, x='AveragePrice', y='Total Volume', color='region')
fig.show()

# Don't stop here! Continue Exploring